xdsl/interactive/app.py

```diff
@@ -147,8 +178,24 @@ def update_current_module(self) -> None:
             pipeline = PipelinePass([p() for p in self.current_selected_pass_list])
             pipeline.apply(ctx, module)
             self.current_module = module
+            # for current_module's that are valid ModuleOp's find the condensed pass list
+            self.current_condensed_pass_list = condensed_pass_list(self.current_module)
+
+            # trigger watch_condense_mode function every time current_module is updated to recalculate/reupdate condensed_pass_list
+            if self.condense_mode is True:
+                self.condense_mode = True
+            else:
+                self.condense_mode = False
```

Comment:
This kinds of code feels very weird. I know, it is supposed to trigger the update-function on a write to the variable, but this kind of "implicit function calling" is not very understandable. Can you not just call the update function directly?

Reply:
Textual recommends doing that as little as possible, instead treating UI as derived from the model, and adds machinery to recompute the UI when the model changes. So I would say we want to stay away from calling the updarte function directly as much as possible. I agree with you that this particular bit of code is confusing, which I think is more related to the fact that there hook to update the available list of passes is missing. I think this PR would benefit from a refactor to set up the reactive hook to recompute the list of available passes when the current module changes.